scripts/apps/authoring/authoring/directives/AuthoringDirective.ts

```diff
@@ -846,40 +846,31 @@ export function AuthoringDirective(
                 $scope.autosave(item);
             };
 
-            Object.values(extensions).forEach((extension) => {
-                if (extension.activationResult?.contributions?.authoring != null) {
-                    $scope.onUpdateFromExtension = extension.activationResult.contributions.authoring.onUpdate;
-                }
-            });
-
-            function performOnUpdateFromExtensionAndAutosave(item, origItem, timeout) {
-                $scope.onUpdateFromExtension($scope.origItem._autosave ?? $scope.origItem, $scope.item)
-                    .then((article) => {
-                        angular.extend($scope.item, article);
-                        var autosavedItem = authoring.autosave(item, origItem, timeout);
-
-                        authoringWorkspace.addAutosave();
-                        initMedia();
-                        updateSchema();
-                        return autosavedItem;
-                    });
-            }
-
             $scope.autosave = function(item, timeout) {
                 $scope.dirty = true;
                 angular.extend($scope.item, item); // make sure all changes are available
                 return coreApplyMiddleware(onChangeMiddleware, {item: $scope.item, original: $scope.origItem}, 'item')
                     .then(() => {
-                        if ($scope.onUpdateFromExtension != null) {
-                            performOnUpdateFromExtensionAndAutosave($scope.item, $scope.origItem, timeout);
-                        } else {
+                        const onUpdateFromExtensions = Object.values(extensions).map(
+                            (extension) => extension.activationResult?.contributions?.authoring)
```

Comment:
If you need the `onUpdate` function, you should get it when mapping - `contributions?.authoring?.onUpdate`. As you do it now, you never check if `onUpdate` is `null`. You only check `contributions.authoring`